File: src/day3/TwoSum2.java
package day3;

public class TwoSum2 {
    public static void main(String[] args) {
        int[] nums = {5, 25, 75};
        int[] ints = twoSum(nums, 100);
        for (int anInt : ints) {

        }

    }

    public static int[] twoSum(int[] numbers, int target) {
        int[] result = new int[2];

        int left = 0;
        int right = numbers.length - 1;

        while (left <= right) {
            if (numbers[left] + numbers[right] > target) {
                right--;
            } else if (numbers[left] + numbers[right] < target) {
                left++;
            } else {
                result[0] = left + 1;
                result[1] = right + 1;
                break;
            }
        }

        return result;
    }
}

File: src/algorithm1/kotlinsource/day3/TwoSum2.kt
package algorithm1.kotlinsource.day3

class TwoSum2 {
    fun twoSum2(numbers: IntArray, target: Int): IntArray {
        var left = 0
        var right = numbers.size - 1

        while (left < right) {
            val expected = numbers[left] + numbers[right]
            if (expected == target) {
                break
            } else if (expected < target) {
                left += 1
            } else {
                right -= 1
            }
        }

        return intArrayOf(left + 1, right + 1)
    }
}

fun main() {

}